### PR TITLE
Voting feature

### PR DESCRIPTION
**Pull request by NaiRobley**
#### What does this PR do?
Add a rating/voting feature to 

#### Description of Task to be completed?
Similar to how [StackOverflow](https://stackoverflow.com/) has a rating system on questions and answers, we'd like a simple way for users to rate a data request that has been received.

#### Screenshots
View of a user who up-voted a data request:
image

View of a user who down-voted a data request:
image

View of a user who has not voted or is not logged in:
image

View of votes on the listing of data requests:
image

### CODE REVIEW

File: ckanext/datarequests/templates/datarequests/base.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
 {% block styles %}
   {{ super() }}
   <link rel="stylesheet" href="/datarequests.css" />
+  <script src="/jquery.js"></script>
```
Comment (DavidLemayian) [inline ckanext/datarequests/templates/datarequests/base.html]: I'm pretty sure jquery is included in CKAN core unless I'm mistaken?

File: ckanext/datarequests/templates/datarequests/base.html [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
+  <script type="text/javascript" src="/show.js"></script>
```
Comment (DavidLemayian) [inline ckanext/datarequests/templates/datarequests/base.html]: This file name can be better descriptive. What does show.js do?

Comment (NaiRobley) [reply ckanext/datarequests/templates/datarequests/base.html]: It handles the voting feature. let me rectify that

File: ckanext/datarequests/templates/datarequests/base.html (hunk 1 above)
Comment (NaiRobley) [reply ckanext/datarequests/templates/datarequests/base.html]: Yes. But it is currently not accessible to the extension thus throws an error without the script defined as done above

**Comment by NaiRobley** [comment]
Regarding the icons, the thumbs icons allow us to display the user's vote option upon retrieval of a data request by either using the outlined version or the filled version as shown in the screenshots above to help differentiate and also give feedback to a user upon voting. It is a matter of design which can be changed if preference is use of arrows instead of thumbs

File: ckanext/datarequests/templates/datarequests/base.html (hunk 1 above)
Comment (DavidLemayian) [reply ckanext/datarequests/templates/datarequests/base.html]: @NaiRobley Might it be an issue with precedence? Meaning we have to load the voting scripts after where jquery is called for CKAN core. If not possible, wait for the window to load?

It is not a good idea to load jquery (or any other scripts) twice.

**Comment by DavidLemayian** [comment]
@NaiRobley I'd say that using thumbs feels a lot like the Facebook "like/dislike" feature and might tie to a bit more emotion that will dissuade more people from voting or sending requests after seeing multiple thumbs-down.

I'd therefore say we go for arrows instead similar to StackOverflow.

**Comment by NaiRobley** [comment]
Ok. Will change to the arrows. 

Also, for clarification, will we show the aggregate rating of the data request or display both values of upvotes or downvotes?

**Comment by DavidLemayian** [comment]
@NaiRobley Let's do the sum. So if it receives more down votes than up it would be a negative number.  Just similar to StackOverflow.

**Comment by NaiRobley** [comment]
Ok. On it

**Comment by NaiRobley** [comment]
@DavidLemayian Changed to the arrows. PR updated.

**Comment by DavidLemayian** [comment]
@NaiRobley Thanks. This looks good. I'll open a separate issue for other changes but let's resolve the conflicts for now an I'll merge this.

**Comment by NaiRobley** [comment]
@DavidLemayian Fixed the merge conflict

**Comment by DavidLemayian** [comment]
@NaiRobley there will be continuous improvements so let's have it as a separate PR.

**Pull request merged**